File: 백준/Silver/10773. 제로/제로.cc
#include <stdio.h>
#include <iostream>
#include <stack>


using namespace std;

int main()
{
    int K; 
    cin >> K;
    
    stack<int> st;
    long long sum = 0;
    for(int i=0; i<K; i++)
    {
        int num;
        cin >> num;
        if(num == 0) 
        {
            sum -= st.top();
            st.pop();
        }
        else
        {
            sum+= num; 
            st.push(num);
        }
    }
    cout << sum;
}

File: 백준/Silver/10773. 제로/제로.cs
using System;
using System.Collections.Generic;

namespace baek10773
{
    class Program
    {
        static void Main(string[] args)
        {
            Stack<int> st = new Stack<int>();
            int k = int.Parse(Console.ReadLine());
            for(int i=0; i< k; i++)
            {
                int num = int.Parse(Console.ReadLine());
                //0일경우 무조건 지운다는 정수 보장
                if(num == 0) st.Pop();
                else st.Push(num);
            }
            int sum = 0;
            while(st.Count != 0)
            {
                sum += st.Pop();
            }
            Console.WriteLine(sum);
        }
    }
}